lib/puppet/indirector/direct_file_server.rb

```diff
@@ -7,9 +7,7 @@ class Puppet::Indirector::DirectFileServer < Puppet::Indirector::Terminus
 
   def find(request)
     return nil unless Puppet::FileSystem.exist?(request.key)
-    instance = model.new(request.key)
-    instance.links = request.options[:links] if request.options[:links]
-    instance
+    path2instance(request, request.key)
```

Comment:
Unlike the other calls to `path2instance`, this one was previously lacking a call to `collect`. This appears safe to do for the direct file server, but is it an operation (stating the source file) that needs to be done (although it is nice to be consistent with the other calls, for sure)?